### PR TITLE
Add SIMD feature skeleton

### PR DESCRIPTION
**Pull request by saik0**
This PR creates an empty SIMD feature and wires it to github actions

TODO:
 - [x] Ensure test build passes CI
 - [x] Manually verify the test `test_simd_conditional_compilation` has run in CI
 - [x] Remove `test_simd_conditional_compilation` test

Verified CI in my fork. See https://github.com/saik0/roaring-rs/runs/4975294592?check_suite_focus=true#step:4:108

### CODE REVIEW

File: README.md
Comment (Kerollmops) [inline README.md:74]:
```suggestion
`std::simd` is only available in Rust nightly.
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Hope one day we will be able to run test on an ARM arch!

**Comment by Kerollmops** [comment]
bors try

**Review by Kerollmops — COMMENTED** [review]
Are you OK with bors only checking stable? I would like to add nightly as possibly faillible (we don't care) WDYT?

**Comment by saik0** [comment]
> Are you OK with bors only checking stable? I would like to add nightly as possibly faillible (we don't care) WDYT?

If we do that, we should pin specific nightly toolchain version, and core_simd commit, to prevent that from breaking our builds

**Review by Kerollmops — APPROVED** [review]
I'll merge this like that, we keep the CI like that for now.

Not sure we want to pin a nightly/commit version as I would prefer that we detect breaking changes in `std::simd`.

bors merge

**Comment by saik0** [comment]
> I'll merge this like that, we keep the CI like that for now.
> 
> Not sure we want to pin a nightly/commit version as I would prefer that we detect breaking changes in `std::simd`.
> 
> bors merge

Do failing builds stop PRs from being able to merge?

**Comment by saik0** [comment]
> > I'll merge this like that, we keep the CI like that for now.
> > Not sure we want to pin a nightly/commit version as I would prefer that we detect breaking changes in `std::simd`.
> > bors merge
> 
> Do failing builds stop PRs from being able to merge?



> > I'll merge this like that, we keep the CI like that for now.
> > Not sure we want to pin a nightly/commit version as I would prefer that we detect breaking changes in `std::simd`.
> > bors merge
> 
> Do failing builds stop PRs from being able to merge?

@Kerollmops pinging for visibility post merge. 🙂

**Comment by Kerollmops** [comment]
I think we can setup bors to just warn us about a failing build on nightly, it implies that we must bors try before bors merge. But we could simply rely on the github CI cross/green dot as we do now!

**Comment by saik0** [comment]
I was more concerned that failing build would halt PR from being merged unless they pass

**Comment by Kerollmops** [comment]
I thing bors rules over (?) the github CI, it will merge even if nightly fails in the github.yml CI file.

**Pull request merged**
